src/debug.ts

```diff
@@ -0,0 +1,220 @@
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "chapel.createDebugConfig",
+      chapel_createDebugConfig
+    )
+  );
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "chapel.createDebugConfigForActiveFile",
+      chapel_createDebugConfigForActiveFile
+    )
+  );
+}
+async function chapel_createDebugConfig() {
+  const provider = getProvider();
+  if (!provider) {
+    return
+  }
+  // TODO: we could do some kind of detection for mason projects, maybe even Make/CMake?
+
+  const name = await vscode.window.showInputBox({ placeHolder: "Enter the name of the debug configuration" });
+  if (!name) {
+    return;
+  }
+  const executable = await vscode.window.showInputBox({ placeHolder: "Enter the path to the executable" });
+  if (!executable) {
+    return;
+  }
+  const { argsArray, envMap } = await getArgsAndEnv();
+
+  const newConfig = provider.createDebugConfig(name, executable, argsArray, envMap);
+  await saveConfig(newConfig);
+}
+
+async function chapel_createDebugConfigForActiveFile() {
+  const provider = getProvider();
+  if (!provider) {
+    return
+  }
+
+  // get the current file
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    vscode.window.showErrorMessage("No active editor found.");
+    return;
+  }
+  const document = editor.document;
+  if (document.languageId !== "chapel") {
+    vscode.window.showErrorMessage("The current file is not a Chapel file.");
+    return;
+  }
+  const executablePath = document.fileName?.split(".").slice(0, -1).join(".");
+  if (!executablePath) {
+    vscode.window.showErrorMessage("Failed to get the file name.");
+    return;
+  }
+  const wsPath = vscode.workspace.workspaceFolders?.[0]?.uri.fsPath;
+  if (!wsPath) {
+    vscode.window.showErrorMessage("No workspace folder found.");
+  }
+  let executableNiceName: string = executablePath.split("/").pop() || executablePath;
+  let executableRelPath: string = executablePath;
+  if (wsPath) {
+    // determine executableRelPath to wsPath
+    executableNiceName = executablePath.replace(wsPath + "/", "");
+    executableRelPath = "${workspaceFolder}/" + executableNiceName;
+  }
+
+  const { argsArray, envMap } = await getArgsAndEnv();
+
+  const newConfig = provider.createDebugConfig(`Debug ${executableNiceName}`, executableRelPath, argsArray, envMap);
+  await saveConfig(newConfig);
+}
+
+
+function getProvider(): DebugProvider | undefined {
+  // determine the provider
+  const provider = getPreferredProvider();
+  if (!provider) {
+    const preferredProviderId = getPreferredDebugProvider();
+    vscode.window.showErrorMessage(`The requested debug provider '${preferredProviderId}' is not installed/supported`);
+    return;
+  }
+
+  if (!provider.isAvailable()) {
+    vscode.window.showErrorMessage(`The requested debug provider '${provider.id}' is not installed. Please install [${provider.name}](command:${provider.getInstallCommand()}) or select another provider.`);
```

Comment:
We only support the one debug provider today. "Select another provider" could be misleading.